pkg/mutation/mutation_function_test.go

```diff
@@ -0,0 +1,234 @@
+package mutation_test
+
+import (
+	"fmt"
+	"testing"
+
+	mutationsv1alpha1 "github.com/open-policy-agent/gatekeeper/apis/mutations/v1alpha1"
+	"github.com/open-policy-agent/gatekeeper/pkg/mutation"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const TestValue = "testValue"
```

Comment:
We should be sure to test that we cannot clobber labels that already exist.

Reply:
Added test for AssignMetadata checking this (TestAssignMetadataDoesNotUpdateExistingLabel)